src/modules/dashboard/components/ActionsPage/ActionsComponents/MintTokenMotion.tsx

```diff
@@ -65,10 +66,22 @@ const MintTokenMotion = ({
   transactionHash,
   initiator,
 }: Props) => {
-  const motionTag = MOTION_TAG_MAP[MotionState.Motion];
-  const passedTag = MOTION_TAG_MAP[MotionState.Passed];
-  const revealTag = MOTION_TAG_MAP[MotionState.Reveal];
-  const objectionTag = MOTION_TAG_MAP[MotionState.Objection];
+  const {
+    motionTag,
+    passedTag,
+    revealTag,
+    failedTag,
+    objectionTag,
+  } = useMemo(() => {
+    return Object.values(MOTION_TAG_MAP).reduce((acc, object) => {
+      const { theme, colorSchema } = object as TagAppearance;
+      acc[object.tagName] = (
+        <Tag text={object.name} appearance={{ theme, colorSchema }} />
+      );
+      return acc;
+    }, {} as any);
+  }, []);
```

Comment:
I was meaning to have this refactored.

But they way I think it should be done is to extract it into a separate sub-component.

But only do it if you have time, otherwise you can leave it for the future :)